### PR TITLE
[SD-4353] - Subscriber and Region Targeted Publishing

### PR DESCRIPTION
**Pull request by akintolga**
- [x] Add `codes` field to subscriber entity for optional selector codes
- [x] Make `products` field in subscribers not compulsory
- [x] Implement subscriber targeted publishing
- [x] Move geo restrictions to Products and implement region targeted publishing 
- [x] Check type of subscriber when publishing an embargoed story
- [x] Clean mappers in IPNews formatter

### CODE REVIEW

File: features/targeted_publishing.feature [hunk 1]
```diff
@@ -0,0 +1,334 @@
...
+      "sequence_num_settings":{"min" : 1, "max" : 10}, "email": "test@test.com",
+      "products": ["1"],
+      "destinations":[{"name":"Test","format": "nitf", "delivery_type":"email","config":{"recipients":"test@test.com"}}]
+    },
+    {
+      "_id": "sub-4",
+      "name":"Wire channel with geo restriction Queensland",
+      "media_type":"media",
+      "subscriber_type": "wire",
+      "geo_restrictions": "Queensland",
```
Comment (mdhaman) [inline features/targeted_publishing.feature]: This `geo_restrictions` is not required on the subcsriber??

**Comment by mdhaman** [comment]
:+1: awesome

File: apps/publish/content/tests.py [hunk 2]
```diff
@@ -587,8 +594,30 @@ def test_queue_transmission_for_wire_channels_with_codes(self):
...
+            sub.pop('products', None)
+            subscriber_service.delete({'_id': sub['_id']})
+
+        subscriber_service.post(self.subscribers)
+
+        subscribers, subscribers_yet_to_receive, subscriber_codes = \
+            EnqueuePublishedService().get_subscribers(doc, SUBSCRIBER_TYPES.WIRE)
+
+        self.assertEqual(0, len(subscribers))
+        self.assertEqual(0, len(subscribers))
```
Comment (petrjasek) [inline apps/publish/content/tests.py]: you really want to be sure it's 0 right? :smile:

**Comment by petrjasek** [comment]
:+1:

File: apps/publish/content/tests.py (hunk 2 above)
Comment (akintolga) [inline apps/publish/content/tests.py]: :smile:  I was trying to "double check"...

**Pull request merged**